### DISCUSSION TITLE
Recommended approach for displaying list of key value pairs?

### DISCUSSION THREAD

**Discussion by orjan** [spectreconsole/spectre.console#1082] [Q&A] [answered]
In html where having the definition list to display key value pair and even key and list of values.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

What are you typically using in this case?

**Comment by patriksvensson** [comment, marked as answer]
@orjan I haven't compiled the example below, but something like this should work.
Instead of `Rows`  your could of course use any `IRenderable`.

```csharp
// Create table
var table = new Table();
table.AddColumn("Key");
table.AddColumn("Value");

// Add a row with key/value
table.AddRow("Foo", "Bar");

// Add a new row with multiple rows as values
table.AddRow(
	new Text("Qux"), 
	new Rows(
		new Text("First"),
		new Text("Second")));

// Render table
console.Write(table);
```

**Reply by orjan** [reply]
Thanks for the quick reply, this lib is really awesome! I looked at the table wasn't able to imagine the column of rows.
I'd probably go with the table approach or trying my luck building something custom ;)

**Comment by orjan** [comment]
I tried out the `Tree` and it was simple to use and I'm getting the data that I need.

<img width="187" alt="image" src="https://user-images.githubusercontent.com/124032/203326083-82c96607-8286-4a92-a416-e57e5ea3e2db.png">
